packages/mock-block-dock/dev/test-block.tsx

```diff
@@ -1,14 +1,44 @@
-import { BlockComponent } from "blockprotocol/react";
+import { BlockComponent, useGraphBlockService } from "@blockprotocol/graph";
 import * as React from "react";
+import { useRef } from "react";
 
 type AppProps = {
   name: string;
 };
 
-export const TestBlock: BlockComponent<AppProps> = ({ entityId, name }) => {
+export const TestBlock: BlockComponent<AppProps> = ({ graph }) => {
+  const {
+    blockEntity: { entityId, properties },
+  } = graph;
+  const blockRef = useRef<HTMLDivElement>(null);
+
+  const { graphService } = useGraphBlockService({ ref: blockRef });
```

Comment:
I'd argue the signature should be more like

```suggestion
  const { graphService } = useGraphBlockService(ref, { callbacks: { /*...*/ } });
```

Feels cleaner in the event you don't want to pass callbacks